src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -94,6 +97,12 @@ public Post updatePost(Long courseId, Post post) {
         existingPost.setContent(post.getContent());
         existingPost.setVisibleForStudents(post.isVisibleForStudents());
         existingPost.setTags(post.getTags());
+
+        if (authorizationCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
+            existingPost.setArchived(post.isArchived());
+            existingPost.setPinned(post.isPinned());
```

Comment:
I see the same problem that @jpbernius mentioned in https://github.com/ls1intum/Artemis/pull/3907#discussion_r694872961

We could just throw a bad request in case anyone tries to update a post with both values set to `true`. Even if the UI does not allow doing so, the API still does (and we use it here and there directly). This avoids problems like #3916.